primefaces/src/main/java/org/primefaces/component/gmap/GMapRenderer.java

```diff
@@ -233,6 +235,46 @@ protected void encodeMarker(FacesContext context, Marker marker) throws IOExcept
         writer.write("})");
     }
 
+    protected void encodeIcon(FacesContext context, Object icon) throws IOException {
+        ResponseWriter writer = context.getResponseWriter();
+        if (icon instanceof String) {
+            writer.write("'" + icon + "'");
+            return;
+        }
+        if (!(icon instanceof Symbol)) {
+            throw new FacesException("Icon must be String or Symbol");
```

Comment:
Lets say "GMap icon must be a String or Symbol" just to make the error a little more specific.